ross/defects/crack.py

```diff
@@ -0,0 +1,509 @@
+                             [Koxy[2,0]	,0           ,0         ,0	        ,Koxy[2,1]	,0          ,Koxy[2,2]          ,0                  ,0      ,0                  ,Koxy[2,3]	,0],
+                             [0	        ,Koyz[2,0]   ,0         ,Koyz[2,1]	,0 ,         0          ,0                  ,Koyz[2,2]          ,0      ,Koyz[2,3]          ,0          ,0],
+                             [0	        ,0	         ,0         ,0	        ,0 ,         0          ,0                  ,0                  ,0      ,0                  ,0          ,0],
+                             [0	        ,Koyz[3,0]	 ,0         ,Koyz[3,1]	,0 ,         0          ,0                  ,Koyz[3,2]          ,0      ,Koyz[3,3]          ,0          ,0],
+                             [Koxy[3,0]	,0           ,0         , 0	        ,Koxy[3,1]  ,0          ,Koxy[3,2]          ,0                  ,0      ,0                  ,Koxy[3,3]  ,0],
+                             [0	        ,0           ,0         , 0	        ,0	        ,0          ,0                  ,0                  ,0      ,0                  ,0	        ,0]])
+        # fmt: on
+        F_CRACK = np.zeros(self.ndof)
+
+        KK_CRACK = self.KK - KK_crack
+        FF_CRACK = (KK_CRACK).dot(self.positionsFis[self.dof_crack])
+        F_CRACK[self.dof_crack] = FF_CRACK
+
+        return FF_CRACK, F_CRACK
+
+    def _gasch(self, ap):
+        # Gasch
+        kme = (self.ko + self.kcx) / 2
+        kmn = (self.ko + self.kcz) / 2
+        kde = (self.ko - self.kcx) / 2
+        kdn = (self.ko - self.kcz) / 2
+
+        kee = kme + (4 / np.pi) * kde * (
+            np.cos(ap)
+            - np.cos(3 * ap) / 3
+            + np.cos(5 * ap) / 5
+            - np.cos(7 * ap) / 7
+            + np.cos(9 * ap) / 9
+            - np.cos(11 * ap) / 11
+            + np.cos(13 * ap) / 13
+            - np.cos(15 * ap) / 15
+            + np.cos(17 * ap) / 17
+            - np.cos(19 * ap) / 19
+            + np.cos(21 * ap) / 21
+            - np.cos(23 * ap) / 23
+            + np.cos(25 * ap) / 25
+            - np.cos(27 * ap) / 27
+            + np.cos(29 * ap) / 29
+            - np.cos(31 * ap) / 31
+            + np.cos(33 * ap) / 33
+            - np.cos(35 * ap) / 35
+        )
+
+        knn = kmn + (4 / np.pi) * kdn * (
+            np.cos(ap)
+            - np.cos(3 * ap) / 3
+            + np.cos(5 * ap) / 5
+            - np.cos(7 * ap) / 7
+            + np.cos(9 * ap) / 9
+            - np.cos(11 * ap) / 11
+            + np.cos(13 * ap) / 13
+            - np.cos(15 * ap) / 15
+            + np.cos(17 * ap) / 17
+            - np.cos(19 * ap) / 19
+            + np.cos(21 * ap) / 21
+            - np.cos(23 * ap) / 23
+            + np.cos(25 * ap) / 25
+            - np.cos(27 * ap) / 27
+            + np.cos(29 * ap) / 29
+            - np.cos(31 * ap) / 31
+            + np.cos(33 * ap) / 33
+            - np.cos(35 * ap) / 35
+        )
```

Comment:
Maybe you could replace all these code lines by doing this:
```python
size = 18
cosine_sum = np.sum([(-1) ** i * np.cos((2 * i + 1) * ap) / (2 * i + 1) for i in range(size)])

kee = kme + (4 / np.pi) * kde * cosine_sum 
knn = kmn + (4 / np.pi) * kdn * cosine_sum 
```
You get a cleaner code and better control over how many terms you would like in your sum, without needing "ctrl+c / ctrl+v each time you add a new term.

Reply:
Yes Rodrigo, it is an excellent alternative indeed!